### ISSUE TITLE
[Feature Request] Replace deprecated Nucleus

### ISSUE THREAD

**Issue by TacoTheDank** [inorichi/tachiyomi#2924]
### Why/User Benefit/User Problem
(explain why this feature should be added)

Nucleus is coded in Java, is deprecated, and hasn't been updated since 2017.

ReKotlin is coded in Kotlin, is recommended in the Nucleus README, and is comparatively newer.

https://github.com/konmik/nucleus
https://github.com/ReKotlin/ReKotlin

**Comment by arkon** [comment]
Are you planning on trying to do it? It's quite the undertaking. I'd also double check what approach 1.x took before starting.

**Comment by TacoTheDank** [comment]
No, I don't have the skills required for this sort of thing. It's why I opened an issue for it.

**Comment by CarlosEsco** [comment]
Besides a complete difference in methodology, this library hasn't been updated in almost a year..

**Comment by TacoTheDank** [comment]
I searched around and found some other interesting things.

[This](https://reduxkotlin.org/) is a Kotlin Redux library supporting multi-platform (including Android). Apparently it's a port of Javascript Redux, so it has the same API. https://github.com/reduxkotlin/redux-kotlin

I found two articles that details how to write your own Redux-in-Kotlin, [here](https://jayrambhia.com/blog/kotlin-redux-architecture) and [here](https://medium.com/swlh/how-to-implement-redux-in-kotlin-part-1-the-basics-db2854613079).

These might be worth looking into.
